src/libcollections/benches/slice.rs

```diff
@@ -290,3 +290,24 @@ sort!(sort_unstable, sort_unstable_large_random, gen_random, 10000);
 sort!(sort_unstable, sort_unstable_large_big_random, gen_big_random, 10000);
 sort!(sort_unstable, sort_unstable_large_strings, gen_strings, 10000);
 sort_expensive!(sort_unstable_by, sort_unstable_large_random_expensive, gen_random, 10000);
+
+macro_rules! reverse {
+    ($name:ident, $ty:ident) => {
+        #[bench]
+        fn $name(b: &mut Bencher) {
+            // odd length and offset by 1 to be as unaligned as possible
+            let n = 0xFFFFF;
+            let mut v: Vec<_> =
+                (0..1+(n / mem::size_of::<$ty>() as u64))
+                .map(|x| x as $ty)
+                .collect();
+            b.iter(|| black_box(&mut v[1..]).reverse());
+            b.bytes = n;
+        }
+    }
+}
+
+reverse!(reverse_u8, u8);
+reverse!(reverse_u16, u16);
+reverse!(reverse_u32, u32);
+reverse!(reverse_u64, u64);
```

Comment:
Makes sense to have all the primitives.  Also added `[u8;3]` and `Simd<[f64;4]>` while I was at it, to show more of the perf range.

Results, from fastest to slowest:
```text
test slice::reverse_simd_f64x4  ... bench:      36,818 ns/iter (+/-   924) = 28479 MB/s
test slice::reverse_u128        ... bench:      41,797 ns/iter (+/- 3,127) = 25087 MB/s
test slice::reverse_u64         ... bench:      47,062 ns/iter (+/-   898) = 22280 MB/s
test slice::reverse_u8          ... bench:      51,678 ns/iter (+/- 3,819) = 20290 MB/s
test slice::reverse_u32         ... bench:      74,404 ns/iter (+/-   387) = 14092 MB/s
test slice::reverse_u16         ... bench:      92,952 ns/iter (+/- 2,385) = 11280 MB/s
test slice::reverse_u8x3        ... bench:     181,223 ns/iter (+/- 6,541) =  5786 MB/s
```